pkg/usersettings/handlers.go

```diff
@@ -0,0 +1,170 @@
+package usersettings
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	core "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+
+	"github.com/openshift/console/pkg/auth"
+	"github.com/openshift/console/pkg/proxy"
+	"github.com/openshift/console/pkg/serverutils"
+)
+
+const namespace = "openshift-console-user-settings"
+
+var USER_RESOURCE = schema.GroupVersionResource{
+	Group:    "user.openshift.io",
+	Version:  "v1",
+	Resource: "users",
+}
+
+type UserSettingsHandler struct {
+	K8sProxyConfig      *proxy.Config
+	Client              *http.Client
+	Endpoint            string
+	ServiceAccountToken string
+}
+
+func (h *UserSettingsHandler) HandleUserSettings(user *auth.User, w http.ResponseWriter, r *http.Request) {
+	context := context.TODO()
+
+	serviceAccountClient, err := h.createServiceAccountClient()
+	if err != nil {
+		klog.Errorf("Failed to create service account to handle user setting request: %v", err)
+		serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to create service account to handle user setting request: %v", err)})
+		return
+	}
+
+	userSettingMeta, err := h.getUserSettingMeta(context, user)
+	if err != nil {
+		klog.Errorf("Failed to get user data to handle user setting request: %v", err)
+		serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to get user data to handle user setting request: %v", err)})
+		return
+	}
+
+	switch r.Method {
+	case http.MethodGet:
+		configMap, err := h.getUserSettings(context, serviceAccountClient, userSettingMeta)
+		if err != nil {
+			klog.Errorf("Failed to get user settings: %v", err)
+			serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to get user settings: %v", err)})
+		} else {
+			serverutils.SendResponse(w, http.StatusOK, configMap)
+		}
+	case http.MethodPost:
+		configMap, err := h.createUserSettings(context, serviceAccountClient, userSettingMeta)
+		if err != nil {
```

Comment:
The preferred idiomatic way for go, how to structure the response should be:
```go
if err != nil {
        errMsg := fmt.Sprintf("Failed to create user settings: %v", err)
	klog.Errorf(errMsg)
	serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: errMsg})
	return
}
serverutils.SendResponse(w, http.StatusOK, configMap)
```
so we return immediatelly

Reply:
done